bin/clover/src/pipeline-steps/assetSpecificOverrides.ts

```diff
@@ -110,6 +110,27 @@ const overrides = new Map<string, OverrideFn>([
     setAnnotationOnSocket(socket, { tokens: ["GroupId"] });
     variant.sockets.push(socket);
   }],
+  ["AWS::EC2::NetworkInterface", (spec: ExpandedPkgSpec) => {
+    const variant = spec.schemas[0].variants[0];
+
+    // Add an annotation for the Id output socket to connect to HostedZoneId
+    const socket = variant.sockets.find(
+      (s: ExpandedSocketSpec) => s.name === "Id" && s.data.kind === "output",
+    );
+    if (!socket) return;
+
+    setAnnotationOnSocket(socket, { tokens: ["NetworkInterfaceId"] });
+
+    const prop = variant.domain.entries.find((p: ExpandedPropSpec) =>
+      p.name === "GroupSet"
+    );
+
+    if (!prop) return;
+    const groupSocket = createInputSocketFromProp(prop);
+
+    setAnnotationOnSocket(groupSocket, { tokens: ["GroupId"] });
+    variant.sockets.push(groupSocket);
```

Comment:
It feels like we'll get more mileage out of these overrides by going the other way and adding `GroupSet` as an annotation on the relevant output socket, especially since these show up more than once